src/system/kernel/ipc/utility.c

```diff
@@ -13,8 +13,46 @@
 #include <hal/proc/proc.h>
 #include <error/panic.h>
 
-bool k_ipc_utl_pathIsNotification(const char * absPathPtr, const size_t pathLen) {
-    return absPathPtr[0] == GNW_PATH_IPC_BROADCAST_PREFIX;
+bool k_ipc_utl_pathGlobalValidate(const char * absPathPtr, const size_t pathLen) {
+    for (size_t i = 0; i < pathLen; ++i) {
+        if (IN_RANGE('0', absPathPtr[i], '9')                          ||   /* 0 - 9 */
+            IN_RANGE('A', absPathPtr[i], 'Z')                          ||   /* A - Z */
+            IN_RANGE('a', absPathPtr[i], 'z')                          ||   /* a - z */
+            ((absPathPtr[i] == GNW_PATH_IPC_COMPONENT_SEPARATOR) && i)      /* path separator */) {
+            continue;
+        }
+
+        return false;
+    }
+
+    return true;
```

Comment:
1. Czy pusty ciąg znaków powinien być poprawny? np. `""`
2. Czy ścieżka składająca się z samych separatorów powinna być poprawna? `"///"`
3. Czy ścieżka zawierająca pusty komponent powinna być poprawna? np. `"aa//bb"`
4. Czy ścieżka zawierająca pusty komponent powinna być tożsama ze ścieżką która takiego komponentu nie zawiera? np. `aa/bb` =? `aa/bb`

Reply:
Słusznie, trochę było napisane po taniości - in progress

Reply:
1 - Gotowe

Reply:
2 - Nie powinna - gotowe
3 - Nie powinna - gotowe
4 - Nie powinna - gotowe w związku z punktem 3